Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,429 @@
+     * @throws \Exception
+     *
+     * @return \Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface|null
+     */
+    public function receiveNoWait()
+    {
+        /*if ($this->currentEnvelope) {
+            throw new \RuntimeException(
+                'AmqpQueueDriver: This driver has a message that was not acknowledged yet. A message must be processed and acknowledged before receiving new messages.'
+            );
+        }
+
+        $this->dequeueingTimeMs = 0;
+
+        $this->currentEnvelope = $this->queue->get();
+
+        $msg = null;
+
+        if (!$this->currentEnvelope) {
+            return null;
+        }
+
+        $start = microtime(true);
+        $deserializationContext = new DeserializationContext();
+
+        try{
+            /** @var QueueMessageInterface $msg */
+//            $msg = $this->getSerializer()->deserialize($this->currentEnvelope->getBody(), SerializableInterface::class, $this->format, $deserializationContext);
+        /*} catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception, ['headers' => $this->currentEnvelope->getHeaders(), 'body' => $this->currentEnvelope->getBody()]);
+            $this->ack();
+            return null;
+        }
+        foreach ($this->currentEnvelope->getHeaders() as $header => $value) {
+            $msg->setHeader($header, $value);
+        }
+
+        // Calculate how long it took to deserilize the message
+        $this->dequeueingTimeMs = (int) ((microtime(true) - $start) * 1000);
+
+        return $msg;*/
+    }
+
+    /**
+     * @todo verify the need of this function
+     * @return int The time it took in ms to de-queue and deserialize the message
+     */
+    public function getDequeueingTimeMs()
+    {
+//        return $this->dequeueingTimeMs;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purge(string $queue)
+    {
+        $this->manager->getQueue($queue)->purge();
+    }
+
+    /**
+     * Sleep the script for a given number of seconds.
+     *
+     * @param int|float $seconds
+     */
+    public function sleep($seconds)
+    {
+        if ($seconds < 1) {
+            usleep((int) $seconds * 1000000);
+        } else {
+            sleep((int) $seconds);
+        }
+    }
+
+    /**
+     * @return PhpAmqpLibQueueManager|QueueManager
+     */
+    public function getManager()
+    {
+        return $this->manager;
+    }
+
+    /**
+     * @param string $queueName
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function setQueue($queueName)
+    {
+        $this->declareQueue($queueName);
+    }
+
+    /**
+     * @param $exchange
+     */
+    public function setExchange($exchange)
+    {
+        $this->declareExchange($exchange);
+    }
+
+    /**
+     * Declares a channel to connect with the broker
+     */
+    public function setChannel()
+    {
+        $this->declareChannel();
```

Comment:
This call is not going to work because `declareChannel` requires some parameters. I also think that this method `setChannel` can be deleted based on the latest changes you did

Reply:
It was removed